File: mario.c
#include <stdio.h>

void getHeight(int *height);
void spaces(int length);
void pyramide(int length);

int main(void)
{
    int height;
    getHeight(&height);

    int height1 = height;
    
    for (int l = 0; l < height; l++)
    {
        spaces(height1 - 1);
        pyramide(l);
        printf("\n");
        height1--;
    }

    return 0;
}

void getHeight(int *height)
{
    do
    {
        printf("Height: ");
        scanf("%i", height);
    }
    while (*height < 1 || *height > 8);
}

void spaces(int length)
{
    for (int i = 0; i < length; i++)
    {
        printf(" ");
    }
}

void pyramide(int length)
{
    for (int j = 0; j < length + 1; j++)
    {
        printf("#");
    }
    printf("  ");
    for (int k = 0; k < length + 1; k++)
    {
        printf("#");
    }
}

File: mario.py
# Prints out a pyramid of asterisks
def spaces(length1):
    for i in range(length1 - 1):
        print(" ", end='')


def pyramide(length):
    for i in range(length):
        print("#", end='')
    print("  ", end='')

    for i in range(length):
        print("#", end='')
    print("\n", end='')


def main():
    height = 0

    while True:
        try:
            height = int(input("Height: "))
            if 1 <= height <= 8:
                break
        except ValueError:
            print("Type an integer between 1 and 8")
    height1 = height
    for i in range(height):
        spaces(height1)
        pyramide(i + 1)
        height1 -= 1


main()
